### PR TITLE
fix(content-structured-data): fix author link

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Refact: move `makeAbsolute` and `isSameLink` utility functions from Storyblok.helpers to a new file `@utils/url.ts`
* Refact: update implementation of `makeAbsolute` function. Before something like `/se/insurances/car-insurance` was being considered an absolute URL just because of the leading forward slash ('/').
* Refact: update `isSameLink` implementation. Now it get's two URL strings and check if they are the same despite of them being absolute or relative URLs.
* Fix: Fixed two issues with content structured data:
  * Make it proper handle absolute or relative URLs given as `authorLink`. For structured data Google expects absolute URLs so I'm using `getLinkFieldURL` to achieve that.
  * Make sure images used for structured data use the same domain we've been using for storyblok assets.

## Justify why they are needed

Request from Peter

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
* **#4421** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4421?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @guilhermespopolin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx [hunk 1]
```diff
@@ -34,11 +36,11 @@ function ArticleStructuredData({ story }: { story: PageStory }) {
           '@context': 'https://schema.org',
           '@type': type,
           ...(headline && { headline }),
-          ...(image && { image: image.filename }),
+          ...(image && { image: getImgSrc(image.filename) }),
```
Comment (guilhermespopolin) [inline apps/store/src/app/[locale]/[[...slug]]/ContentTypeStructuredData.tsx:39]: Request from Peter. Make sure we don't include a new domain for images.

File: apps/store/src/utils/url.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+export const makeAbsolute = (url: string) => {
+  if (/^(?:[a-zA-Z]+:)?\/\//.test(url)) {
+    return url
+  }
+  return new URL(url, ORIGIN_URL).href
+}
+
+export const isSameLink = (urlA: string, urlB: string) => {
+  return makeAbsolute(urlA) === makeAbsolute(urlB)
+}
```
Comment (alebedev) [inline apps/store/src/utils/url.ts:13]: <i class='graphite__hidden'>[Re: line 14]<br/><br/></i>

Great to have separate file!  Maybe things like `removeTrailingSlash` should be here as well?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4421?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: apps/store/src/services/storyblok/Storyblok.helpers.ts [hunk 3]
```diff
@@ -43,13 +44,6 @@ export const getLinkFieldURL = (link: LinkField, linkText?: string) => {
...
 }
 
-export const makeAbsolute = (url: string) => {
-  if (/^(\/|https?:\/\/|\/\/)/.test(url)) {
-    return url
-  }
-  return '/' + url
-}
-
 const appendAnchor = (url: string, anchor?: string) => (anchor ? `${url}#${anchor}` : url)
```
Comment (alebedev) [inline apps/store/src/services/storyblok/Storyblok.helpers.ts:47]: This too looks like URL util

**Review by alebedev — APPROVED** [review]

File: apps/store/src/utils/url.ts (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/utils/url.ts:13]: True. Have pushed a new PR to this stack with some more changes.

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jun 28, 8:06 AM EDT**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4421).

**Pull request merged**